lbrynet/core/HTTPBlobDownloader.py

```diff
@@ -0,0 +1,79 @@
+from random import choice
+import logging
+
+from twisted.internet import defer
+import treq
+from twisted.internet.task import LoopingCall
+
+log = logging.getLogger(__name__)
+
+
+class HTTPBlobDownloader(object):
+    def __init__(self, blob_manager, blob_hashes=None, servers=None, client=None):
+        self.blob_manager = blob_manager
+        self.servers = servers or []
+        self.client = client or treq
+        self.blob_hashes = blob_hashes or []
+        self.looping_call = LoopingCall(self._download_next_blob_hash_for_file)
+        self.failures = 0
+        self.max_failures = 3
+        self.interval = 1
+
+    @property
+    def running(self):
+        return self.looping_call.running
+
+    def start(self):
+        if not self.running and self.blob_hashes and self.servers:
+            return self.looping_call.start(self.interval, now=True)
+        defer.succeed(None)
+
+    def stop(self):
+        if self.running:
+            self.blob_hashes = []
+            return self.looping_call.stop()
+
+    @defer.inlineCallbacks
+    def _download_next_blob_hash_for_file(self):
+        for blob_hash in self.blob_hashes:
+            blob = yield self.blob_manager.get_blob(blob_hash)
+            if not blob.verified:
+                self.download_blob(blob)
+                return
+        self.stop()
+
+    @defer.inlineCallbacks
+    def download_blob(self, blob):
+        try:
+            yield self._download_blob(blob)
+            self.failures = 0
+        except Exception as exception:
+            self.failures += 1
+            log.error('Mirror failed downloading: %s', exception)
```

Comment:
If you want the traceback you can do `log.exception("Mirror failed downloading")`